include/deal.II/base/bounding_box.h

```diff
@@ -193,6 +267,61 @@ class BoundingBox
   std::pair<Point<spacedim, Number>, Point<spacedim, Number>> boundary_points;
 };
 
+
+/**
+ * Returns the unit box [0,1]^dim.
+ */
+template <int dim, typename Number = double>
+BoundingBox<dim, Number>
+create_unit_box();
+
+
+/**
+ * This function defines a convention for how coordinates in dim
+ * dimensions should translate to the coordinates in dim + 1 dimensions,
+ * when one of the coordinates in dim + 1 dimensions is locked to a given
+ * value.
+ *
+ * The convention is the following: Starting from the locked coordinate we
+ * store the lower dimensional coordinates consecutively and wrapping
+ * around when going over the dimension. This relationship can be
+ * described by the following tables:
+ *
+ *                        2D
+ * ------------------------------------------------
+ * | locked in 2D | 1D coordinate | 2D coordinate |
+ * |--------------------------------------------- |
+ * |     x0       |      (a)      |   (x0,  a)    |
+ * |     x1       |      (a)      |   (a , x1)    |
+ * ------------------------------------------------
+ *
+ *                       3D
+ * --------------------------------------------------
+ * | locked in 3D | 2D coordinates | 3D coordinates |
+ * |----------------------------------------------- |
+ * |     x0       |    (a, b)      | (x0,  a,  b)   |
+ * |     x1       |    (a, b)      | ( b, x1,  a)   |
+ * |     x2       |    (a, b)      | ( a,  b, x2)   |
+ * --------------------------------------------------
+ *
+ * Given a locked coordinate, this function maps a coordinate index in dim
+ * dimension to a coordinate index in dim + 1 dimensions.
+ *
+ * @param locked_coordinate should be in the range [0, dim+1).
+ * @param coordiante_in_dim should be in the range [0, dim).
+ * @return A coordinate index in the range [0, dim+1)
+ */
+template <int dim>
+inline unsigned int
+coordinate_to_one_dim_higher(const unsigned int locked_coordinate,
+                             const unsigned int coordiante_in_dim)
```

Comment:
Is this a function of general interest? Otherwise, it might just go into the `.cc` file or an anonymous namespace.

Same with `@relates` as above.

Reply:
No, I can't see that it would be useful externally. But for the functionality I'm working on, I will need it internally in more places than in this class, so I would like to keep it in the header. I have moved it to an internal namespace in #9852 .